XpringKit/JavaScript/JavaScriptSigner.swift

```diff
@@ -0,0 +1,46 @@
+import Foundation
+import JavaScriptCore
+
+/// Provides Signer functionality backed by JavaScript.
+internal class JavaScriptSigner {
+    /// String constants which refer to named JavaScript resources.
+    private enum ResourceNames {
+        public static let signer = "Signer"
+        public static let signTransaction = "signTransaction"
+    }
+
+    /// A JavaScriptSerializer which can convert native objects to JavaScript.
+    private let javaScriptSerializer: JavaScriptSerializer
+
+    /// Native JavaScript functions wrapped by this class.
+    private let signTransactionFunction: JSValue
+
+    /// Initialize a new Signer.
+    ///
+    /// - Note: Initialization will fail if the expected bundle is missing or malformed.
+    public init() {
+        let context = XRPJavaScriptLoader.XRPJavaScriptContext
+
+        let signer = XRPJavaScriptLoader.load(ResourceNames.signer, from: context)
+        signTransactionFunction = XRPJavaScriptLoader.load(ResourceNames.signTransaction, from: signer)
+
+        javaScriptSerializer = JavaScriptSerializer(context: context)
+    }
+
+    /// Sign a transaction.
+    ///
+    /// - Parameters:
+    ///    - transaction: The `Transaction` to sign.
+    ///    - wallet: The wallet which will sign the transaction.
+    /// - Returns: A `SignedTransaction` derived from the inputs.
+    public func sign(_ transaction: Rpc_V1_Transaction, with wallet: Wallet) -> [UInt8]? {
+        guard let javaScriptTransaction = javaScriptSerializer.serialize(transaction: transaction) else {
+            return nil
+        }
+        let javaScriptWallet = javaScriptSerializer.serialize(wallet: wallet)
+
+        let javaScriptSignedBytes = signTransactionFunction.call(withArguments: [javaScriptTransaction, javaScriptWallet])!
+        let signedBytes = javaScriptSignedBytes.toArray()
+        return signedBytes as? [UInt8]
```

Comment:
What's the `as?` here?

Reply:
This needed master merged into it so this is already committed. 

To answer your question:
`as` is a cast operator. You can have:
- `as`: perform a cast that we know will always succeed due to static types (ie. `0 as Integer`)
- `as?`: attempt a cast and if the cast fails, assign nil:
```swift
let x = 0 as? Integer // x = 0, type Integer? (Optional integer)
let y = 0.5 as? Integer // y = 0, type Integer? (Optional integer)
```
- `as!`: force cast, and crash if things don't go great. This is generally poor form to use in code:
```swift
let x = 0 as! Integer // x = 0, type Integer 
let y = 0.5 as? Integer // program will crash with runtime exception. 
```

Reply:
Neat. So this works differently than `as` in TypeScript, because Swift has real types.

In TypeScript, the `as` operator tells the compiler to treat the value as that type, but it doesn't really perform a cast or do anything else.